File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/ErdOnlineRoleService.java
package org.javamaster.invocationlab.admin.service;

import org.javamaster.invocationlab.admin.enums.RoleGroupEnum;
import org.javamaster.invocationlab.admin.model.erd.PermissionResVo;
import org.javamaster.invocationlab.admin.model.erd.RolePermissionResVo;
import org.javamaster.invocationlab.admin.model.erd.RoleResVo;
import org.javamaster.invocationlab.admin.model.erd.SaveCheckedOperationsReqVo;
import org.javamaster.invocationlab.admin.model.erd.TokenVo;
import org.javamaster.invocationlab.admin.model.erd.UsersResVo;
import com.alibaba.fastjson.JSONObject;
import org.apache.commons.lang3.tuple.Pair;

import java.util.List;
import java.util.Set;

/**
 * @author yudong
 */
public interface ErdOnlineRoleService {
    RolePermissionResVo getUserSubRoles(String projectId, TokenVo tokenVo) throws Exception;

    List<RoleResVo> getRoleGroups(String projectId, TokenVo tokenVo);

    UsersResVo getRoleGroupUsers(String projectId, String roleId, Integer current, Integer pageSize, TokenVo tokenVo) throws Exception;

    UsersResVo searchUsers(String projectId, String username, TokenVo tokenVo) throws Exception;

    UsersResVo addUsersToRoleGroup(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    UsersResVo delUsersFromRoleGroup(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    PermissionResVo getRoleGroupSubRoles(String projectId, String roleId, TokenVo tokenVo) throws Exception;

    String saveCheckedOperations(SaveCheckedOperationsReqVo reqVo, TokenVo tokenVo);

    void saveRoleGroupSubRolesToRedis(String projectId, Set<Long> roleIds, RoleGroupEnum roleGroupEnum);

    Pair<Set<Long>, RoleGroupEnum> getUserRoleIds(String userId, String projectId) throws Exception;

    RoleGroupEnum getUserRoleGroup(String userId, String projectId);

    void saveRolesWhenCreateProject(String projectId, TokenVo tokenVo);

    void delRolesWhenDelProject(String projectId, TokenVo tokenVo);
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/ErdOnlineRoleService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.enums.RoleGroupEnum
import org.javamaster.invocationlab.admin.model.erd.*
import com.alibaba.fastjson.JSONObject
import org.apache.commons.lang3.tuple.Pair

/**
 * @author yudong
 */
interface ErdOnlineRoleService {
    fun getUserSubRoles(projectId: String, tokenVo: TokenVo): RolePermissionResVo

    fun getRoleGroups(projectId: String, tokenVo: TokenVo): List<RoleResVo>

    fun getRoleGroupUsers(
        projectId: String,
        roleId: String,
        current: Int,
        pageSize: Int,
        tokenVo: TokenVo
    ): UsersResVo

    fun searchUsers(projectId: String, username: String, tokenVo: TokenVo): UsersResVo

    fun addUsersToRoleGroup(jsonObjectReq: JSONObject, tokenVo: TokenVo): UsersResVo

    fun delUsersFromRoleGroup(jsonObjectReq: JSONObject, tokenVo: TokenVo): UsersResVo

    fun getRoleGroupSubRoles(projectId: String, roleId: String, tokenVo: TokenVo): PermissionResVo

    fun saveCheckedOperations(reqVo: SaveCheckedOperationsReqVo, tokenVo: TokenVo): String

    fun saveRoleGroupSubRolesToRedis(projectId: String, roleIds: Set<Long>, roleGroupEnum: RoleGroupEnum)

    fun getUserRoleIds(userId: String, projectId: String): Pair<Set<Long>, RoleGroupEnum>

    fun getUserRoleGroup(userId: String, projectId: String): RoleGroupEnum?

    fun saveRolesWhenCreateProject(projectId: String, tokenVo: TokenVo)

    fun delRolesWhenDelProject(projectId: String, tokenVo: TokenVo)
}
